### PR TITLE
fix(horizontalNav): update active state of selected nav item

### PR DESCRIPTION
**Pull request by varnikajain15**
### What does this implement/fix? Explain your changes.

...

### Does this close any currently open issues?

...

### Any other comments?

...

### Dependent PRs/Commits

...

### Describe breaking changes, if any.

...

### Checklist

Check all those that are applicable and complete.

- [ ] Merged with latest `master` branch

### CODE REVIEW

File: core/components/organisms/horizontalNav/HorizontalNav.tsx [hunk 1]
```diff
@@ -48,12 +48,7 @@ export const HorizontalNav = (props: HorizontalNavProps) => {
 
     if (menu.icon) {
       return (
-        <Icon
-          className="mr-3 HorizontalNav-animate"
-          name={menu.icon}
-          appearance={getIconAppearance(isActive, menu.disabled)}
-          data-test="DesignSystem-HorizontalNav--Icon"
-        />
+        <Icon className="mr-3 HorizontalNav-animate" name={menu.icon} data-test="DesignSystem-HorizontalNav--Icon" />
```
Comment (Rashi-Gupta-2000) [inline core/components/organisms/horizontalNav/HorizontalNav.tsx:51]: Please do not replace appearance with CSS but just override the need CSS.

File: css/src/components/horizontalNav.css [hunk 2]
```diff
@@ -16,6 +16,11 @@
 
 .HorizontalNav-menu--disabled {
   pointer-events: none;
+  color: var(--inverse-lightest);
```
Comment (Rashi-Gupta-2000) [inline css/src/components/horizontalNav.css]: Please keep this in the JS file so that there is no redundant work.

**Review by Rashi-Gupta-2000 — DISMISSED** [review]

File: css/src/components/horizontalNav.css [hunk 3]
```diff
@@ -16,6 +16,11 @@
...
+}
+
+.HorizontalNav-menu--default {
+  color: var(--inverse);
```
Comment (anuradha9712) [inline css/src/components/horizontalNav.css]: We are using the same colours for MenuText as well. Please create a common function to get colours for both MenuWrapper and MenuText and reuse it.

File: core/components/organisms/horizontalNav/HorizontalNav.tsx [hunk 4]
```diff
@@ -62,13 +57,15 @@ export const HorizontalNav = (props: HorizontalNavProps) => {
 
   const list = menus.map((menu, index) => {
     const isActive = isMenuActive(menus, menu, active);
+    const color = getTextColor(isActive, menu.disabled);
```
Comment (anuradha9712) [inline core/components/organisms/horizontalNav/HorizontalNav.tsx]: Please change the function name and variable name

**Review by anuradha9712 — CHANGES_REQUESTED** [review]
Please add a test cases to check the correct classes used based on various conditions

**Review by anuradha9712 — APPROVED** [review]

**Pull request merged**